scripts/e2e-test.sh

```diff
@@ -51,7 +51,14 @@ helm template deploy/helm/eunomia-operator/ \
   --set eunomia.operator.deployment.enabled= \
   --set eunomia.operator.namespace=$TEST_NAMESPACE | kubectl apply -f -
 
-operator-sdk test local ./test/e2e --namespace "$TEST_NAMESPACE" --up-local --no-setup --verbose --go-test-flags "-tags e2e -timeout 20m"
+go test ./test/e2e/... \
+  -namespacedMan /dev/null \
+  -globalMan /dev/null \
+  -root "$EUNOMIA_PATH" \
+  -test.parallel=1 \
+  -localOperator \
+  -tags e2e \
+  -v
```

Comment:
I'm curious on this one too. Can we ask the `operator-sdk` maintainers if there is a way to make this work? Maybe we can open an issue or post in their slack channel?

Reply:
I cannot run `operator-sdk test local`'s `--up-local` flag without the `--namespace` flag:
https://github.com/operator-framework/operator-sdk/blob/v0.8.1/cmd/operator-sdk/test/local.go#L116-L117 (it's also the case in the latest operator-sdk).
<details><summary><code>operator-sdk test local --help</code></summary><br><pre>
$ operator-sdk test local --help
Run End-To-End tests locally

Usage:
  operator-sdk test local <path to tests directory> [flags]

Flags:
      --debug                        Enable debug-level logging
      --global-manifest string       Path to manifest for Global resources (e.g. CRD manifests)
      --go-test-flags string         Additional flags to pass to go test
  -h, --help                         help for local
      --image string                 Use a different operator image from the one specified in the namespaced manifest
      --kubeconfig string            Kubeconfig path
      --molecule-test-flags string   Additional flags to pass to molecule test
      --namespace string             If non-empty, single namespace to run tests in
      --namespaced-manifest string   Path to manifest for per-test, namespaced resources (e.g. RBAC and Operator manifest)
      --no-setup                     Disable test resource creation
      --up-local                     Enable running operator locally with go run instead of as an image in the cluster

Global Flags:
      --verbose   Enable verbose logging
</pre></details>
However, it's still possible to run the tests successfully by issuing `operator-sdk test local ./test/e2e --no-setup --verbose --go-test-flags "-tags e2e -timeout 20m"` (deploying eunomia to the cluster beforehand) **but** then all the tests are being run in the `default` namespace (and that's because of [GetNamespace() function](https://github.com/operator-framework/operator-sdk/blob/v0.8.1/pkg/test/resource_creator.go#L36-L39) - `*singleNamespace` is in turn enforced by [`--no-setup` flag](https://github.com/operator-framework/operator-sdk/blob/v0.8.1/cmd/operator-sdk/test/local.go#L205-L207)).

**Finally**
Well, in course of writing this comment I actually found the right `oprator-sdk`'s flag combination to run the tests successfully in separate namespaces (no need to resort to bare `go test ...`), but eunomia has to be deployed to the cluster beforehand (I cannot use `--up-local`) but I think this is fine, because apart from e2e tests we now have a deployment test at the same time (so we don't need a [separate test](https://github.com/KohlsTechnology/eunomia/blob/master/scripts/e2e-test.sh#L60-L80) at the end of the `e2e-test.sh` script - and if the deployment fails, e2e tests just won't run):
```
operator-sdk test local ./test/e2e \
--namespaced-manifest /dev/null \
--global-manifest /dev/null \
--verbose \
--go-test-flags "-tags e2e -timeout 20m"
```
Soooo... Thank you for the comments, they helped me a lot! 👍